Fix what seems to be a mismerge of a couple patches relating to StackFrameLists…

jimingham opened the pull request:
Felipe noticed some code in StackFrameList::ResetCurrentInlinedDepth that didn't make sense and wasn't present upstream (anymore).  It looks like in the course of cherry-picking patches and their reverts and revert-reverts this code got a little mangled.  This patch fixes the cherry-picks up to llvm.org PR #150718 - which hadn't been applied yet.

jimingham committed:
Fix what seems to be a mismerge of a couple patches relating to StackFrameLists
to the stable/21.x branch.

felipepiovezan reviewed (approved):
Thanks for investigating this!

jimingham commented:
@swift-ci please test

jimingham committed:
Fix two calls to GetSuggestedStackFrameIndex that weren't honoring
the inline_stack argument.  This was causing the test failure in
TestSwiftRuntimeInstrumentationRecognizer.py.

jimingham committed:
Merge branch 'stable/21.x' into fixup-stack-writer-mutex-cherrypick

jimingham commented:
@swift-ci please test
